### PR TITLE
513_remove_deprecated: remove functions and parameters deprecated bef…

### PR DESCRIPTION
**Pull request by bundfussr**
…ore 0.6.0

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://roche-gsk.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/) 
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://roche-gsk.github.io/admiral/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://roche-gsk.github.io/admiral/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::documents()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md`
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://roche-gsk.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue so that it closes after successful merging. 
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done!  Much love to your accomplishment!

### CODE REVIEW

**Review by bms63 — COMMENTED** [review]
Should we say something in the news file about these functions being fully deprecated and removed from admiral?

**Comment by bundfussr** [comment]
Yes, I have updated the news file.

File: NEWS.md [hunk 1]
```diff
@@ -1,22 +1,45 @@
...
 
-- The `derive_last_dose()` function has been split into a general function 
-`derive_vars_last_dose()`, and three wrapper functions `derive_var_last_dose_amt()`, 
-`derive_var_last_dose_date()`, and `derive_var_last_dose_grp()` (#385)
+- The following functions and parameters, which were deprecated in previous admiral versions, were removed:
+  
+  - `derive_aage()` function
+  - `derive_duration()` function
+  - `derive_query_vars()` function
+  - `derive_suppqual_vars()` function
```
Comment (bms63) [inline NEWS.md:31]: Can these just be grouped on one line?  News file is getting long.

File: NEWS.md [hunk 2]
```diff
@@ -1,22 +1,45 @@
...
+  - `fns` and `filter_rows` parameter in `derive_summary_records()`
+  - `date_var` and `traceabilty_vars` parameter in `dthcaus_source()`
```
Comment (bms63) [inline NEWS.md:33]: should parameter be plural?  parameters?

File: NEWS.md (hunk 1 above)
Comment (bundfussr) [reply NEWS.md]: Updated

File: NEWS.md (hunk 2 above)
Comment (bundfussr) [reply NEWS.md:40]: I would consider "`fns` and `filter_rows` parameter" as short form of "the `fns` parameter and the `filter_rows` parameter". However, I am not a native speaker. So I am not sure if this is correct. I would be really interested in an answer as this comes up frequently when writing documentation.

@pharmaverse/core-team , can anyone answer this grammar question?

Comment (rossfarrugia) [reply NEWS.md:40]: "parameters" would work here as Ben suggests

**Pull request merged**